static/app/app/controller.js

```diff
@@ -59,6 +60,34 @@ var Controller = Marionette.Object.extend({
             spots.fetch().done(callback);
         }
 
+    },
+    join: function(moveId) {
+        var moves = channel.request('entities:moves');
+        
+        $.when(moves.fetch()).done(function(){
+            var currentMove = channel.request("entities:move");
+            var moveToJoin = channel.request('entities:moves')
+                                    .find({id: parseInt(moveId)});
```

Comment:
Backbone offers a specialized method for retrieving items by id, [get](http://backbonejs.org/#Collection-get). I think get might not require you to coerce the string to an integer either.